packages/moonstone/Scrollable/Scrollbar.js

```diff
@@ -18,131 +18,121 @@ import componentCss from './Scrollbar.module.less';
-class ScrollbarBase extends Component {
-	static displayName = 'ScrollbarBase'
-
-	static propTypes = /** @lends moonstone/Scrollable.Scrollbar.prototype */ {
-		/**
-		 * Called when [ScrollThumb]{@link moonstone/Scrollable.ScrollThumb} is updated.
-		 *
-		 * @type {Function}
-		 * @private
-		 */
-		cbAlertThumb: PropTypes.func,
-
-		/**
-		 * Client size of the container; valid values are an object that has `clientWidth` and `clientHeight`.
-		 *
-		 * @type {Object}
-		 * @property {Number}    clientHeight    The client height of the list.
-		 * @property {Number}    clientWidth    The client width of the list.
-		 * @public
-		 */
-		clientSize: PropTypes.shape({
-			clientHeight: PropTypes.number.isRequired,
-			clientWidth: PropTypes.number.isRequired
-		}),
-
-		/**
-		 * Adds the corner between vertical and horizontal scrollbars.
-		 *
-		 * @type {Booelan}
-		 * @default false
-		 * @public
-		 */
-		corner: PropTypes.bool,
-
-		/**
-		 * `true` if rtl, `false` if ltr.
-		 * Normally, [Scrollable]{@link ui/Scrollable.Scrollable} should set this value.
-		 *
-		 * @type {Boolean}
-		 * @private
-		 */
-		rtl: PropTypes.bool,
-
-		/**
-		 * Registers the ScrollButtons component with an
-		 * {@link core/internal/ApiDecorator.ApiDecorator}.
-		 *
-		 * @type {Function}
-		 * @private
-		 */
-		setApiProvider: PropTypes.func,
-
-		/**
-		 * The scrollbar will be oriented vertically.
-		 *
-		 * @type {Boolean}
-		 * @default true
-		 * @public
-		 */
-		vertical: PropTypes.bool
-	}
-
-	static defaultProps = {
-		corner: false,
-		vertical: true
-	}
-
-	constructor (props) {
-		super(props);
-
-		if (props.setApiProvider) {
-			props.setApiProvider(this);
-		}
-
-		this.scrollbarRef = React.createRef();
-		this.scrollButtonsRef = React.createRef();
-	}
-
-	componentDidMount () {
-		const {getContainerRef, showThumb, startHidingThumb, update: uiUpdate} = this.scrollbarRef.current;
-
-		this.getContainerRef = getContainerRef;
-		this.showThumb = showThumb;
-		this.startHidingThumb = startHidingThumb;
-		this.uiUpdate = uiUpdate;
-
-		const {isOneOfScrollButtonsFocused, updateButtons, focusOnButton} = this.scrollButtonsRef.current;
-
-		this.isOneOfScrollButtonsFocused = isOneOfScrollButtonsFocused;
-		this.update = (bounds) => {
-			updateButtons(bounds);
-			this.uiUpdate(bounds);
+const ScrollbarBase = memo(forwardRef((props, ref) => {
+	// Refs
+	const scrollbarRef = useRef();
+	const scrollButtonsRef = useRef();
+	// render
+	const {cbAlertThumb, clientSize, corner, vertical, ...rest} = props;
+
+	useImperativeHandle(ref, () => {
+		const {getContainerRef, showThumb, startHidingThumb, update: uiUpdate} = scrollbarRef.current;
+		const {isOneOfScrollButtonsFocused, updateButtons, focusOnButton} = scrollButtonsRef.current;
+		return {
+			getContainerRef,
+			showThumb,
+			startHidingThumb,
+			uiUpdate,
+			isOneOfScrollButtonsFocused,
+			update: (bounds) => {
+				updateButtons(bounds);
+				uiUpdate(bounds);
+			},
+			focusOnButton
```

Comment:
When creating a new PR, it would be better to order the properties alphabetically.

Reply:
👌